src/object-store/src/layers/lru_cache/read_cache.rs

```diff
@@ -237,24 +236,24 @@ impl<C: Accessor + Clone> ReadCache<C> {
     {
         OBJECT_STORE_LRU_CACHE_MISS.inc();
 
-        let inner_result = inner.read(path, args).await;
+        // The real read will happen after `read_to_end` is invoked.
+        let (_, mut reader) = inner.read(path, args).await?;
+        let mut buf = Vec::new();
```

Comment:
Looks like we know the size, could we initialize the buf with a proper capacity?

Reply:
I back it to the original implementation to avoid `read_to_end` reading all data into memory.